common/pkg/capi/clustertopology/handlers/mutation/meta.go

```diff
@@ -108,20 +127,88 @@ func (mgp metaGeneratePatches) GeneratePatches(
 
 			log.V(3).Info("Starting mutation pipeline", "handlerCount", len(mgp.mutators))
 
-			for i, h := range mgp.mutators {
-				handlerName := fmt.Sprintf("%T", h)
-				log.V(5).Info("Running mutator", "index", i, "handler", handlerName)
+			// Merge the global variables to the current resource vars. This allows the handlers to access
+			// the variables defined in the cluster class or cluster configuration and use these correctly when
+			// overrides are specified on machine deployment or control plane configuration.
+			mergedVars, err := mergeVariableDefinitions(vars, globalVars)
+			if err != nil {
+				log.Error(err, "Failed to merge global variables")
+				return err
+			}
 
-				if err := h.Mutate(ctx, obj.(*unstructured.Unstructured), vars, holderRef, clusterKey, clusterGetter); err != nil {
-					log.Error(err, "Mutator failed", "index", i, "handler", handlerName)
+			for i, h := range mgp.mutators {
+				mutatorType := fmt.Sprintf("%T", h)
+				log.V(5).
+					Info("Running mutator", "index", i, "handler", mutatorType, "vars", vars)
+
+				if err := h.Mutate(
+					ctx,
+					obj.(*unstructured.Unstructured),
+					mergedVars,
+					holderRef,
+					clusterKey,
+					clusterGetter,
+				); err != nil {
+					log.Error(err, "Mutator failed", "index", i, "handler", mutatorType)
 					return err
 				}
 
-				log.V(5).Info("Mutator completed successfully", "index", i, "handler", handlerName)
+				log.V(5).Info("Mutator completed successfully", "index", i, "handler", mutatorType)
 			}
 
 			log.V(3).Info("Mutation pipeline completed successfully", "handlerCount", len(mgp.mutators))
 			return nil
 		},
 	)
 }
+
+func mergeVariableDefinitions(
+	vars, globalVars map[string]apiextensionsv1.JSON,
```

Comment:
```suggestion
	overrideVars, globalVars map[string]apiextensionsv1.JSON,
```

Can you use a name that's better reflects what this function is intending to do? A comment here would also be helpful.

Reply:
Done.